### PR TITLE
Project: Members Only

### PR DESCRIPTION
**Pull request by LamiaSristy**
This project deals with the authentication and authorization of users.
We have used the 'devise' gem for the authentication and made the model user using the gem.
We created the controller for new, create and index methods and corresponding views.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<h1>All Possts</h1>
+
+<ul id="posts">
+  <% @posts.each do |post| %>
+    <li>
+      <%= post.title %>
+      <%= post.body %>
+      <% if user_signed_in?  %>    
+         <%= post.user.email %>
+      <%end%>
```
Comment (maelfosso) [inline app/views/posts/index.html.erb]: No logic in view helpers, move them in view helpers, please.

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    </li>
+  <% end %>
+</ul>
+
+<% if user_signed_in?  %>    
+    <%= link_to "Create a New Post", new_post_path %>
+<%end%>
```
Comment (maelfosso) [inline app/views/posts/index.html.erb]: No logic in view helpers, move them in view helpers, please.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -10,6 +10,23 @@
...
   <body>
+    <p class="navbar-text float-right">
+      <% if user_signed_in? %>
+        Logged in as <strong><%= current_user.email %></strong>.
+        <%= link_to 'Edit profile', edit_user_registration_path, :class => 'navbar-link' %> |
+        <%= link_to "Logout", destroy_user_session_path, method: :delete, :class => 'navbar-link'  %>
+      <% else %>
+        <%= link_to "Sign up", new_user_registration_path, :class => 'navbar-link'  %> |
+        <%= link_to "Login", new_user_session_path, :class => 'navbar-link'  %>
+      <% end %>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb]: No logic in view helpers, move them in view helpers, please.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required 🔴 

Hello,

Thank you for the well-done job. It's working like a charm.
However, there is some missing requirement

1- Please, the `rubocop.yml` file is different from the one required. Kindly follow this [link](https://github.com/microverseinc/linters-config/tree/master/ruby) to set up the good one

2- Read the comments left below, please.

Kindly make the suggested changes and once done submit a new code review request
Thanks

Mael FOSSO

File: app/helpers/posts_helper.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+module PostsHelper
+  def check_signed
+    if user_signed_in?
+      # <%= link_to "Create a New Post", new_post_path %>
```
Comment (mcrd25) [inline app/helpers/posts_helper.rb]: REMOVE COMMENTED OUT CODE

**Review by mcrd25 — APPROVED** [review]
# changes approved :heavy_check_mark: 

Hi Team!

Thanks for making the changes requested! no need for further review!

. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya Douglas](mayacdouglas.com)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Pull request merged**